synapse/lib/stormtypes.py

```diff
@@ -530,43 +531,34 @@ def addLibFuncs(self):
         })
 
     async def _methQueueAdd(self, name):
-
         self.runt.reqAllowed(('storm', 'queue', 'add'))
-
-        info = self.runt.snap.core.multiqueue.queues.get(name)
-        if info is not None:
-            mesg = f'A queue named {name} already exists.'
-            raise s_exc.DupName(mesg=mesg)
-
-        info = {'user': self.runt.user.iden, 'time': s_common.now()}
-        self.runt.snap.core.multiqueue.add(name, info)
-
+        info = await self.runt.snap.addCoreQueue(name, {})
         return Queue(self.runt, name, info)
 
     async def _methQueueGet(self, name):
 
-        info = self.runt.snap.core.multiqueue.queues.get(name)
+        info = await self.runt.snap.getCoreQueue(name)
         if info is None:
             mesg = f'No queue named {name}.'
             raise s_exc.NoSuchName(mesg=mesg, name=name)
 
         return Queue(self.runt, name, info)
 
-    async def _methQueueDel(self, name, allow=()):
+    async def _methQueueDel(self, name):
 
-        info = self.runt.snap.core.multiqueue.queues.get(name)
-        if info is None:
+        if not await self.runt.snap.hasCoreQueue(name):
             mesg = f'No queue named {name} exists.'
             raise s_exc.NoSuchName(mesg=mesg, name=name)
 
-        if info.get('user') != self.runt.user.iden:
-            self.runt.reqAllowed(('storm', 'queue', 'del', name))
+        info = await self.runt.snap.getCoreQueue(name)
 
-        await self.runt.snap.core.multiqueue.rem(name)
+        if (info.get('user') == self.runt.user.iden or
+            self.runt.reqAllowed(('storm', 'queue', 'del', name))):
+            await self.runt.snap.delCoreQueue(name)
 
     async def _methQueueList(self):
-        self.runt.reqAllowed(('storm', 'lib', 'queue', 'list'))
-        return self.runt.snap.core.multiqueue.list()
+        self.runt.reqAllowed(('storm', 'queue', 'list'))
```

Comment:
this is going to be a breaking change.

Reply:
I can revert it, but perhaps we should discuss.  I was thinking we should (where possible) use a generic perms convention rather than specific to the lib layout.

Reply:
Please revert that in this branch and we can do a separate pass because there are other permissions to clean up; as well as potential namespace collisions with the storm queue hierarchy that is out of scope for this delta.